### ISSUE TITLE
询问一下预训练的问题

### ISSUE THREAD

**Issue by Nine9844** [bubbliiiing/unet-pytorch#22]
你好，打扰了。我是想问下主干模型是指的是在下采样过程中使用的vgg吗？如果我不改变上采样是不是就不用使用imagenet训练。然后注销掉model_path=‘’ 以及 if model_path !=‘’这段。然后使用自己的数据集去进行训练。  谢谢大佬！！！！！！。实际上大佬你的voc的权重文件是不是为二次预训练的数据。
不好意思，语言表达能力不行。俺不晓得这样说大佬明不明白。

**Comment by Nine9844** [comment]
我还想问一下为什么dice的计算是需要进行    temp_inputs = torch.softmax(inputs.transpose(1, 2).transpose(2, 3).contiguous().view(n, -1, c),-1)       # 在这里为什么使用了softmax
    temp_target = target.view(n, -1, ct)
    #--------------------------------------------#
    #   计算dice系数
    #--------------------------------------------#
    temp_inputs = torch.gt(temp_inputs, threhold).float()
    tp = torch.sum(temp_target[...,:-1] * temp_inputs, dim=[0,1])  #想知道这里为什么是这样计算。有推荐的书籍看吗？谢谢
    fp = torch.sum(temp_inputs                       , dim=[0,1]) - tp
    fn = torch.sum(temp_target[...,:-1]              , dim=[0,1]) - tp

**Comment by bubbliiiing** [comment]
1、逻辑看不懂，我感觉没有逻辑关系，你想从0开始也行
2、因为要求概率才能算dice

**Comment by Nine9844** [comment]
大佬对不起（＞人＜；）不好意思，前天没组织好语言。
1、主干模型指的是unet中使用的vgg和resnet吗？
2、您训练出来的voc.pth权重文件是否是冻结主干训练出来的。我不想使用voc的pth，想使用其他的数据训练
3、想知道dim为啥取[0，1]，取这个是获取图像的什么。
4、大佬大佬，有关语义分割的书籍推荐吗？这种tensor变换真的看不明白。
谢谢大佬！！！！！！！！！！！！！！！！！！！！！

**Comment by bubbliiiing** [comment]
1、是
2、从主干开始训练获得
3、取所有图的所有像素点
4、没有……语义分割的逻辑很简单了已经，就像素点分个类

**Comment by Nine9844** [comment]
谢谢大佬
